Flagsmith.FlagsmithClient/Cache/FlagListCache.cs

```diff
@@ -12,19 +12,24 @@ public abstract class FlagListCache
 
         protected IFlags _flags;
         protected readonly IDateTimeProvider _dateTimeProvider;
-        protected DateTime _timestamp;
+        protected DateTime? _timestamp;
 
         protected FlagListCache(IDateTimeProvider dateTimeProvider, IFlags flags, int cacheDurationInMinutes)
         {
             _flags = flags;
             _dateTimeProvider = dateTimeProvider;
-            _timestamp = dateTimeProvider.Now();
+            _timestamp = null;
             _cacheDurationInMinutes = cacheDurationInMinutes;
         }
 
         protected bool IsCacheStale()
         {
-            return _dateTimeProvider.Now().Subtract(_timestamp).TotalMinutes > _cacheDurationInMinutes;
+            if (_timestamp == null)
+            {
+                return true;
+            }
+
+            return _dateTimeProvider.Now().Subtract((DateTime)_timestamp).TotalMinutes > _cacheDurationInMinutes;
```

Comment:
This is a personal (pythonic) preference, but I would rather see something like: 

```csharp
return _timestamp == null || _dateTimeProvider.Now().Subtract((DateTime)_timestamp).TotalMinutes > _cacheDurationInMinutes;
```

(maybe this is not the correct ternary form here but something like this seems cleaner in my opinion but this isn't necessary for approval)

Reply:
Good idea! Will update this in a minute

Reply:
fixed